app/mobile-content/pages/edit-page-dialog/edit-page-dialog.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, Inject, OnInit, DoCheck } from '@angular/core';
+import { MdDialogRef, MD_DIALOG_DATA } from '@angular/material';
+
+import { MobilePage, pageType } from '../shared/mobile-page.model';
+
+@Component({
+  moduleId: module.id,
+  selector: 'hip-edit-page-dialog',
+  templateUrl: 'edit-page-dialog.component.html'
+})
+export class EditPageDialogComponent implements OnInit, DoCheck {
+  page: MobilePage;
+  private prevType: pageType;
+
+  constructor(public dialogRef: MdDialogRef<EditPageDialogComponent>,
+              @Inject(MD_DIALOG_DATA) public data: { pageToEdit: MobilePage }) {}
+
+  ngOnInit() {
+    if (this.data && this.data.pageToEdit) {
+      // deep clone input page to make editing cancelable
+      this.page = MobilePage.parseObject(JSON.parse(JSON.stringify(this.data.pageToEdit)));
```

Comment:
Before writing this line, I did my own research on how to most efficiently deep clone an object in JS and `JSON.parse(JSON.stringify)` was the shortest and most efficient. Unfortunately, in this case the parsed object also needs to be explicitly of type `MobilePage`, otherwise you get an error. So it needs to be parsed again. If I knew a better way, I'd use it there.